modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/DomainRoutes.scala

```diff
@@ -24,27 +27,25 @@ import scala.concurrent.{ExecutionContext, Future}
 /**
   * Http route definitions for domain specific functionality.
   *
-  * @param domains       the domain operation bundle
-  * @param querySettings query parameters form settings
-  * @param queryBuilder  query builder for domains
-  * @param base          the service public uri + prefix
+  * @param domains           the domain operation bundle
+  * @param domainQueries     query builder for domains
+  * @param base              the service public uri + prefix
+  * @param querySettings     query parameters from settings
+  * @param filteringSettings filtering parameters from settings
   */
-final class DomainRoutes(domains: Domains[Future], querySettings: QuerySettings, queryBuilder: FilterQueries[DomainId], base: Uri) {
+final class DomainRoutes(domains: Domains[Future], domainQueries: FilterQueries[Future, DomainId], base: Uri)(implicit querySettings: QuerySettings, filteringSettings: FilteringSettings) {
 
   private val encoders = new DomainCustomEncoders(base)
-  import encoders._, queryBuilder._
-
-  private val pagination = querySettings.pagination
+  import encoders._
 
   def routes: Route = handleExceptions(ExceptionHandling.exceptionHandler) {
     handleRejections(RejectionHandling.rejectionHandler) {
       pathPrefix("organizations" / Segment / "domains") { orgIdString =>
         val orgId = OrgId(orgIdString)
-        pathEndOrSingleSlash {
-          (get & parameter('from.as[Int] ? pagination.from) & parameter('size.as[Int] ? pagination.size) & parameter('deprecated.as[Boolean].?)) { (from, size, deprecated) =>
-            traceName("listDomains") {
-              queryBuilder.listingQuery(orgId, deprecated, None, Pagination(from, size)).response
-            }
+        (pathEndOrSingleSlash & get & searchQueryParams) { (pagination, filterOpt, termOpt, deprecatedOpt) =>
+          traceName("listDomains") {
```

Comment:
should we rename to `filterDomains` instead? dunno yet. @MFSY ?

Reply:
I like the prefix 'search'. It's independent from the modality (filter, full text search) used to access the data

Reply:
searchDomains
searchOrganizations
searchSchemas
searchInstances